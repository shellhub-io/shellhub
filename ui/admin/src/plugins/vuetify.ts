// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";
import { VNumberInput } from "vuetify/labs/VNumberInput";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
  components: {
    VNumberInput,
  },
});
