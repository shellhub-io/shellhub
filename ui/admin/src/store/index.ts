import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";
import { stats, StatsState } from "./modules/stats";

import { UsersState, users } from "./modules/users";
import { AuthState, auth } from "./modules/auth";
import { DevicesState, devices } from "./modules/devices";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { layout, LayoutState } from "./modules/layout";
import { license, LicenseState } from "./modules/license";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { sessions, SessionsState } from "./modules/sessions";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { spinner, SpinnerState } from "./modules/spinner";
import { announcement, AnnouncementState } from "./modules/announcement";
import { instance, InstanceState } from "./modules/instance";

export interface State {
  user: UsersState;
  auth: AuthState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  license: LicenseState;
  namespaces: NamespacesState;
  publicKeys: PublicKeysState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  announcement: AnnouncementState;
  instance: InstanceState;
}

// eslint-disable-next-line symbol-description
export const key: InjectionKey<Store<State>> = Symbol();

export const store = createStore<State>({
  modules: {
    users,
    auth,
    devices,
    firewallRules,
    layout,
    license,
    namespaces,
    publicKeys,
    sessions,
    snackbar,
    spinner,
    stats,
    announcement,
    instance,
  },
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
