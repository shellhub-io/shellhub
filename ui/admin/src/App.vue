<template>
  <v-app>
    <component :is="layout" :data-test="`${layout}-component`" />
  </v-app>
</template>

<script lang="ts">
import { defineComponent, onMounted, computed } from "vue";
import { useRouter } from "vue-router";
import SimpleLayout from "./layouts/SimpleLayout.vue";
import AppLayout from "./layouts/AppLayout.vue";
import { useStore } from "./store";
import { INotificationsError } from "./interfaces/INotifications";

export default defineComponent({
  name: "App",
  components: {
    appLayout: AppLayout,
    simpleLayout: SimpleLayout,
  },
  setup() {
    const store = useStore();
    const router = useRouter();

    const layout = computed(() => store.getters["layout/getLayout"]);

    const isLoggedIn = computed(() => store.getters["auth/isLoggedIn"]);

    const currentRoute = computed(() => router.currentRoute.value.path);

    onMounted(async () => {
      if (!isLoggedIn.value) {
        try {
          await store.dispatch("auth/logout");
          store.dispatch("layout/setLayout", "simpleLayout");
          router.push("/login");
        } catch {
          store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.namespaceLoad);
        }
      }

      if (isLoggedIn.value && currentRoute.value !== "/login") {
        const license = await store.dispatch("license/get");
        if (!license || license.expired) {
          store.dispatch("snackbar/showSnackbarErrorAction", INotificationsError.license);
          store.dispatch("layout/setLayout", "appLayout");
          router.push("/license");
        }
        store.dispatch("layout/setLayout", "appLayout");
      }
    });

    return {
      layout,
    };
  },
});
</script>
