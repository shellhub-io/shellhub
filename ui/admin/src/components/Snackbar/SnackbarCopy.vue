<template>
  <v-snackbar
    v-model="snackbar"
    :timeout="2000"
    location="top"
    color="#F9F3EE"
    variant="tonal"
    transition="slide-x-transition"
  >
    {{ message }}
  </v-snackbar>
</template>

<script setup lang="ts">
import { computed } from "vue";
import { useStore } from "../../store";

const props = defineProps({
  mainContent: {
    type: String,
    default: "",
    required: true,
  },
});
const store = useStore();

const snackbar = computed({
  get() {
    return store.getters["snackbar/snackbarCopy"];
  },
  set() {
    store.dispatch("snackbar/unsetShowStatusSnackbarCopy");
  },
});

const message = computed(() => `${props.mainContent} copied to clipboard.`);
</script>
