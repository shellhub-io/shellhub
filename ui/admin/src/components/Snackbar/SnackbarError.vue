<template>
  <v-snackbar
    v-model="snackbar"
    :timeout="4000"
    color="#bd4147"
    location="top"
    transition="slide-x-transition"
  >
    {{ message }}
  </v-snackbar>
</template>

<script setup lang="ts">
import { computed } from "vue";
import { useStore } from "../../store";

const props = defineProps({
  typeMessage: {
    type: String,
    required: true,
  },

  mainContent: {
    type: String,
    default: "",
    required: false,
  },
});
const store = useStore();

const snackbar = computed({
  get() {
    return store.getters["snackbar/snackbarError"];
  },
  set() {
    store.dispatch("snackbar/unsetShowStatusSnackbarError");
  },
});

const message = computed(() => {
  switch (props.typeMessage) {
    case "custom":
      return props.mainContent;
    case "loading":
      return `Loading the ${props.mainContent} has failed, please try again.`;
    case "action":
      return `The ${props.mainContent} request has failed, please try again.`;
    case "licenseRequired":
      return `The ${props.mainContent} request has failed, license required.`;
    default:
      return "The request has failed, please try again.";
  }
});

defineExpose({ snackbar, message });
</script>
