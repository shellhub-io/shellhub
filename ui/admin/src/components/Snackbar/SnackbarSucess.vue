<template>
  <v-snackbar
    v-model="snackbar"
    location="top"
    :timeout="4000"
    color="#4caf50"
    variant="tonal"
    transition="slide-x-transition"
  >
    <p class="w-100 text-center">{{ message }}</p>
  </v-snackbar>
</template>

<script setup lang="ts">
import { computed } from "vue";
import { useStore } from "../../store";

const props = defineProps({
  typeMessage: {
    type: String,
    required: true,
  },

  mainContent: {
    type: String,
    default: "",
    required: false,
  },
});
const store = useStore();

const snackbar = computed({
  get() {
    return store.getters["snackbar/snackbarSuccess"];
  },
  set() {
    store.dispatch("snackbar/unsetShowStatusSnackbarSuccess");
  },
});

const message = computed(() => {
  switch (props.typeMessage) {
    case "action":
      return `The ${props.mainContent} has succeeded.`;
    case "no-content":
      return "There is no content to export";
    default:
      return "The request has succeeded.";
  }
});

defineExpose({ snackbar, message });
</script>
