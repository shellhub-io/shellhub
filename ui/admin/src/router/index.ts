import { createRouter, createWebHistory } from "vue-router";
import Login from "@admin/views/Login.vue";
import Dashboard from "@admin/views/Dashboard.vue";
import Users from "@admin/views/Users.vue";
import SettingsLicense from "@admin/components/Settings/SettingsLicense.vue";
import SettingsAuthentication from "@admin/components/Settings/SettingsAuthentication.vue";
import Namespaces from "@admin/views/Namespaces.vue";
import Settings from "@admin/views/Settings.vue";

import { store } from "../store";

const routes = [
  {
    path: "/login",
    name: "login",
    component: Login,
  },
  {
    path: "/",
    name: "dashboard",
    component: Dashboard,
  },
  {
    path: "/users",
    name: "users",
    component: Users,
  },
  {
    path: "/user/:id",
    name: "userDetails",
    component: () => import("@admin/views/UserDetails.vue"),
  },
  {
    path: "/devices",
    name: "devices",
    component: () => import("@admin/views/Device.vue"),
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: () => import("@admin/views/Device.vue"),
      },
    ],
  },
  {
    path: "/device/:id",
    name: "deviceDetails",
    component: () => import("@admin/views/DeviceDetails.vue"),
  },
  {
    path: "/sessions",
    name: "sessions",
    component: () => import("@admin/views/Sessions.vue"),
  },
  {
    path: "/session/:id",
    name: "sessionDetails",
    component: () => import("@admin/views/SessionDetails.vue"),
  },
  {
    path: "/settings",
    name: "Settings",
    component: Settings,
    redirect: { name: "SettingProfile" },
    children: [
      {
        path: "authentication",
        name: "SettingAuthentication",
        component: SettingsAuthentication,
      },
      {
        path: "license",
        name: "SettingLicense",
        component: SettingsLicense,
      },
    ],
  },
  {
    path: "/firewall-rules",
    name: "firewall-rules",
    component: () => import("@admin/views/FirewallRules.vue"),
  },
  {
    path: "/firewall-rules/:id",
    name: "firewallRulesDetails",
    component: () => import("@admin/views/FirewallRulesDetails.vue"),
  },
  {
    path: "/namespaces",
    name: "namespaces",
    component: Namespaces,
  },
  {
    path: "/namespace/:id",
    name: "namespaceDetails",
    component: () => import("@admin/views/NamespaceDetails.vue"),
  },
  {
    path: "/announcements",
    name: "announcements",
    component: () => import("@admin/views/Announcements.vue"),
  },
  {
    path: "/announcement/:uuid",
    name: "announcementDetails",
    component: () => import("@admin/views/AnnouncementDetails.vue"),
  },
  {
    path: "/new-announcement",
    name: "new-announcement",
    component: () => import("@admin/views/NewAnnouncement.vue"),
  },
];

const router = createRouter({
  history: createWebHistory("/admin/"),
  routes,
});

router.beforeEach((to, from, next) => {
  if (to.path !== "/login") {
    if (store.getters["auth/isLoggedIn"]) {
      return next();
    }
    return next(`/login?redirect=${to.path}`);
  }
  if (store.getters["auth/isLoggedIn"]) {
    if (to.path === "/login" && to.query.token) {
      return next();
    }
    return next("/");
  }

  return next();
});

export default router;
