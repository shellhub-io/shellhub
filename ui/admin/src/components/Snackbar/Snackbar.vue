<template>
  <snackbar-sucess
    :type-message="message.typeMessage"
    :main-content="message.typeContent"
  />

  <snackbar-error
    :type-message="message.typeMessage"
    :main-content="message.typeContent"
  />

  <snackbar-copy
    :main-content="message.typeContent"
  />
</template>

<script setup lang="ts">
import { computed } from "vue";
import SnackbarSucess from "./SnackbarSucess.vue";
import SnackbarError from "./SnackbarError.vue";
import SnackbarCopy from "./SnackbarCopy.vue";
import { useStore } from "../../store";

const store = useStore();

const message = computed(() => store.getters["snackbar/snackbarMessageAndContentType"]);

defineExpose({ message });
</script>
