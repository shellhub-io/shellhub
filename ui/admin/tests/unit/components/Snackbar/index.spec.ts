import { createStore } from "vuex";
import { createVuetify } from "vuetify";
import { shallowMount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it } from "vitest";
import Snackbar from "../../../../src/components/Snackbar/Snackbar.vue";
import { key } from "../../../../src/store";
import router from "../../../../src/router";

const snackbarMessageAndContentType = {
  typeMessage: "",
  typeContent: "",
};
const store = createStore({
  state: {
    snackbarMessageAndContentType,
  },
  getters: {
    "snackbar/snackbarMessageAndContentType": (state) => state.snackbarMessageAndContentType,
  },
  actions: {},
});

describe("Device Icon", () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    const vuetify = createVuetify();

    wrapper = shallowMount(Snackbar, {
      global: {
        plugins: [[store, key], vuetify, router],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper).toBeTruthy();
  });
  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("Process data in the computed", () => {
    expect(wrapper.vm.message).toEqual(snackbarMessageAndContentType);
  });
});
