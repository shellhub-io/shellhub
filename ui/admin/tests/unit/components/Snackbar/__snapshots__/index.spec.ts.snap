// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Device Icon > Renders the component 1`] = `
"<snackbar-sucess-stub typemessage="" maincontent=""></snackbar-sucess-stub>
<snackbar-error-stub typemessage="" maincontent=""></snackbar-error-stub>
<snackbar-copy-stub maincontent=""></snackbar-copy-stub>"
`;
